pydantic/__init__.py

```diff
@@ -399,10 +399,13 @@ def __getattr__(attr_name: str) -> object:
     from importlib import import_module
 
     if module_name == '__module__':
-        return import_module(f'.{attr_name}', package=package)
+        result = import_module(f'.{attr_name}', package=package)
     else:
         module = import_module(module_name, package=package)
-        return getattr(module, attr_name)
+        result = getattr(module, attr_name)
```

Comment:
```suggestion
        result = getattr(module, attr_name)
        g = globals()
        for k, v in module.__dict__:
            if not k.startswith('_'):
                g[k] = v
        return result
```

Reply:
This would ensure we don't re-import the module for everything we grab for it; this could be relevant e.g. if a user's app uses a lot of stuff defined in `pydantic.types` but imported from `pydantic`, I think.